### DISCUSSION TITLE
Nodejs 14.17 Memory Issue

### DISCUSSION THREAD

**Discussion by hshockley** [carbon-design-system/carbon#9011] [Help]
Recently encounted this issue after updating my fork to the most recent changes which included upgrading to Nodejs LTS 14.17

`yarn build` fails primarily when attempting to build carbon-components-react giving the following error

`FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory`

the fix that I found for this was to set the `NODE_OPTIONS` environment variable as done here:
https://stackoverflow.com/questions/38558989/node-js-heap-out-of-memory

This works on solving the issue on my vm which I do development but does not solve the issue when the build is being done in CircleCI

Has anyone else encountered this issue and know of a solution for CircleCI?

**Comment by joshblack** [comment]
Hi there @hshockley! 👋 

Sorry that you're running into memory issues with building `carbon-components-react` 🤔  This isn't something that we've run into on CircleCI/GitHub Actions yet or on our machines. Could you share more about your development environment?
